subject.py

```diff
@@ -0,0 +1,178 @@
+import datajoint as dj
+
+import reference
+
+schema = dj.schema(dj.config['names.%s' % __name__], locals())
+
+
+
+@schema
+class Species(dj.Manual):
+    # <class 'subjects.models.Species'>
+    definition = """
+    species_id:			int             # species id
+    ---
+    binomial:			varchar(255)	# binomial
+    display_name:		varchar(255)	# display name
+    """
+
+
+@schema
+class Strain(dj.Manual):
+    # <class 'subjects.models.Strain'>
+    definition = """
+    strain_id:			int             # strain id
+    ---
+    descriptive_name:		varchar(255)	# descriptive name
+    description:    		varchar(255)	# description
+    """
+
+
+@schema
+class Sequence(dj.Manual):
+    # <class 'subjects.models.Sequence'>
+    definition = """
+    sequence_id:		int             # sequence id
+    ---
+    base_pairs:			varchar(255)	# base pairs
+    description:		varchar(255)	# description
+    informal_name:		varchar(255)	# informal name
+    """
+
+
+@schema
+class Allele(dj.Manual):
+    # <class 'subjects.models.Allele'>
+    definition = """
+    allele_id:			int             # allele id
+    ---
+    standard_name:		varchar(255)	# standard name
+    informal_name:		varchar(255)	# informal name
+    """
+
+
+@schema
+class Line(dj.Manual):
+    # <class 'subjects.models.Line'>
+    # TODO: autoname index: basically autogenerates a unique sequence
+    definition = """
+    -> Species
+    -> Strain
+    -> Sequence
+    line_id:				int             # line id
+    ---
+    name:				varchar(255)	# name
+    description:			varchar(255)	# description
+    target_phenotype:			varchar(255)	# target phenotype
+    auto_name:				varchar(255)	# auto name
+    is_active:				boolean		# is active
+    subject_autoname_index:		integer		# subject autoname index
```

Comment:
I have no idea what these `auto_name` fields refer to...

Reply:
this had something to do with tracking indexes into key generation code logic; possibly can just be zapped